app/Http/Livewire/Components/Market/CreateListing.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Http\Livewire\Components\Market;
+
+use App\Models\Trade;
+use Livewire\Component;
+
+class CreateListing extends Component
+{
+    public $trade_type;
+    public $hydrogen_type;
+    public $units_per_hour;
+    public $duration;
+    public $price_per_unit;
+    public $mix_co2;
+    public $expires_at;
+
+    // Default select does not work with livewire at the moment:
+    public $duration_type = "day";
+    public $expires_at_type = "day";
+
+    protected $rules = [
+        'trade_type' =>     'required',
+        'hydrogen_type' =>  'required',
+        'units_per_hour' => 'required|numeric|min:0|max:1000000',
+        'duration' =>       'required|numeric',
+        'price_per_unit' => 'required|numeric|min:0|max:1000000',
+        'mix_co2' =>        'required|numeric|min:0|max:100',
+        'expires_at' =>     'required|numeric',
+    ];
+
+    public function render()
+    {
+        return view('livewire.components.market.create-listing');
+    }
+
+    public function submit()
+    {
+        $data = $this->validate();
+
+        // Add owner_id value to data
+        $data['owner_id'] = auth()->id();
+
+        // Modify duration data
+        $duration = $data['duration'] * 24;
+
+        if($this->duration_type == 'week' || $this->duration_type == 'month')
```

Comment:
waarom month ook hier?

Reply:
Goed kijken naar de code.
Een maand is in dit geval 24 uur * 7 dagen (vandaar dat die bij week ook is) * 4 week